From 1ca65c37e3da4f95cb0bcf08cb47dfd0a440ab65 Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Tue, 16 Oct 2018 02:16:04 -0700
Subject: [PATCH 0597/1200] mtd: mtdparts: Fix a crash that occurs if
 mtdparts=mtdparts=

If the environment variable mtdparts is equal to "mtdparts=" then U-Boot will crash
because dev is never initialized and it is later freed since it contains garbage.

signed-off-by: Aaron Williams <aaron.williams@marvell.com>
---
 cmd/mtdparts.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/mtdparts.c b/cmd/mtdparts.c
index 9bc977450c..0411babef3 100644
--- a/cmd/mtdparts.c
+++ b/cmd/mtdparts.c
@@ -1550,7 +1550,7 @@ static const char *env_get_mtdparts(char *buf)
 static int parse_mtdparts(const char *const mtdparts)
 {
 	const char *p;
-	struct mtd_device *dev;
+	struct mtd_device *dev = NULL;
 	int err = 1;
 	char tmp_parts[MTDPARTS_MAXLEN];
 
-- 
2.22.0

